lib/pf/scan/wmi/rules.pm

```diff
@@ -0,0 +1,241 @@
+package pf::scan::wmi::rules;
+
+=head1 NAME
+
+pf::scan::wmi::rules - Test wmi rules
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::scan::wmi::rules deny or allow based on rules.
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+use Net::WMIClient qw(wmiclient);
+use Config::IniFiles;
+use pf::api::jsonrpcclient;
+use pf::ConfigStore::WMI();
+
+our %RULE_OPS = (
+    is => sub { $_[0] eq $_[1] ? 1 : 0  },
+    is_not => sub { $_[0] ne $_[1] ? 1 : 0  },
+    match => sub { $_[0] =~ $_[1] ? 1 : 0  },
+    match_not => sub { $_[0] !~ $_[1] ? 1 : 0  },
+);
+
+=head1 SUBROUTINES
+
+=over
+
+=item new
+
+=cut
+
+sub new {
+   my $logger = Log::Log4perl::get_logger("pf::scan::wmi::rules");
+   $logger->debug("instantiating new pf::scan::wmi::rules");
+   my ( $class, %argv ) = @_;
+   my $self = bless {}, $class;
+   return $self;
+}
+
+=item test
+
+Test all the rules
+
+=cut
+
+sub test {
+    my ($self, $rules) = @_;
+    my $logger = Log::Log4perl::get_logger( ref($self) );
+    my $wmics = pf::ConfigStore::WMI->new();
+
+    foreach my $rule  ( @{$rules->{'_rules'}} ) {
+        $logger->warn($rule);
+        my $rule_config = $wmics->read($rule);
+        my ($rc, $result) = $self->runWmi($rules,$rule_config);
+        return $rc if (!$rc);
+        my $action = join("\n", @{$rule_config->{'action'}});
+        my %cfg;
+        tie %cfg, 'Config::IniFiles', ( -file => \$action );
+        foreach my $test  ( sort keys %cfg ) {
+            if ($test =~ /^\w+:(.*)$/) {
+                my $condition = $1;
+                $condition =~ s/(\w+)/$self->parse($cfg{$1},$result)/gee;
+                $condition =~ s/\|/ \|\| /g;
+                $condition =~ s/\&/ \&\& /g;
+                $logger->warn($condition);
+                if (eval $condition) {
+                    $logger->info("Match WMI ".$rule." rule: ".$test." for ". $rules->{'_scanMac'});
+                    if ( defined($cfg{$test}->{'action'}) && $cfg{$test}->{'action'} ne '' ) {
+                        last if ($cfg{$test}->{'action'} =~ /allow/i);
+                        $self->dispatchAction($cfg{$test},$rules,shift @$result);
+                    }
+                }
+            }
+        }
+    }
+    return 0;
+}
+
+=item runWMI
+
+execute WMI command on the remote device
+
+=cut
+
+sub runWmi {
+    my ($self, $rules, $rule) = @_;
+
+    my $request = {};
+    $request->{'Username'} = $rules->{'_domain'} .'/'. $rules->{'_username'} .'%'. $rules->{'_password'};
+    $request->{'Host'} = $rules->{'_scanIp'};
+    $request->{'Query'} = $rule->{'request'};
+    my ($rc, $ret_string) = wmiclient($request);
+    return ($rc,$self->parseResult($ret_string));
+
+}
+
+=item parseResult
+
+Parse the result of the wmicli
+
+=cut
+
+sub parseResult {
+    my ($self, $string) = @_;
+    $string =~ s/\r\n/\n/g;
+
+    my @ans = split('\n', $string);
```

Comment:
@ans ? Is it for answer or for 

![image](https://cloud.githubusercontent.com/assets/3857942/6638605/1e73c3c4-c95e-11e4-98d2-0968b11d1a2e.png)

I guess we could rename that @answer if it's the first case